.github/workflows/enforce-rebase.yml

```
@@ -0,0 +1,30 @@
+name: Enforce Rebase
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  check-rebase:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout PR
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check merge base
+        run: |
+          git fetch origin main
+          # Ensure PR branch is directly on top of main
+          if [ "$(git merge-base HEAD origin/main)" != "$(git rev-parse origin/main)" ]; then
+            echo "Branch is not rebased on latest main. Run `git fetch`, `git rebase origin/main` and `git push --force` to rebase."
```

i think briefly mention that they need to resolve merge conflicts using the Merge Editor and then continuously `git rebase --continue` until successfuly rebase onto remote main, then git push --force

```suggestion
            echo "Branch is not rebased on latest main. Run `git fetch`, `git rebase origin/main` and then resolve all merge conflicts with the Merge Editor. Continue the rebase with `git rebase --continue` until your rebase has completed successfully. Finish with `git push --force` to rebase."
```

idk smth like that